### PR TITLE
[web-browser][ios] Add `preferEphemeralSession` auth session option

### PR DESCRIPTION
**Pull request by barthap**
# Why

Resolves ENG-4526.

Adds support for this option: https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/3237231-prefersephemeralwebbrowsersessio

# How

Changed the options parameter of `openAuthSessionAsync()` to support auth-specific native options and implemented the additional boolean there. I'm not sure how to reword the docs tho.

# Test Plan

NCL

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-web-browser/ios/WebAuthSession.swift [hunk 1]
```diff
@@ -31,6 +31,9 @@ final internal class WebAuthSession {
         ])
       }
     )
+    if #available(iOS 13.0, *) {
+      self.authSession?.prefersEphemeralWebBrowserSession = options.preferEphemeralSession
+    }
```
Comment (barthap) [inline packages/expo-web-browser/ios/WebAuthSession.swift:36]: Maybe I should move this to the `open()` func right below 🤔

Comment (tsapeta) [reply packages/expo-web-browser/ios/WebAuthSession.swift:36]: up to you 🤷‍♂️

File: packages/expo-web-browser/src/WebBrowser.types.ts [hunk 2]
```diff
@@ -84,6 +84,22 @@ export type WebBrowserOpenOptions = {
...
+ * [`WebBrowserOpenOptions`](#webbrowseropenoptions) will be used in the browser polyfill. Otherwise, the browser parameters will be ignored.
+ */
+export type AuthSessionOpenOptions = WebBrowserOpenOptions & {
+  /**
+   * Determines whether the session should ask the browser for a private authentication session.
+   * Set this to `true` to request that the browser doesn’t share cookies or other browsing data between the authentication session and the user’s normal browser session.
+   * Whether the request is honored depends on the user’s default web browser.
+   *
+   * @default false
+   * @platform ios 13+
```
Comment (tsapeta) [inline packages/expo-web-browser/src/WebBrowser.types.ts:98]: Does the `13+` display well? have you tried?

**Review by tsapeta — APPROVED** [review]

File: packages/expo-web-browser/src/WebBrowser.types.ts (hunk 2 above)
Comment (barthap) [reply packages/expo-web-browser/src/WebBrowser.types.ts:98]: Yes, tested, also had a chat with Simek about that (it works but it's fragile)

**Review by bbarthec — APPROVED** [review]

**Pull request merged**